python/ray/air/execution/actor_manager.py

```diff
@@ -0,0 +1,244 @@
+
+        Events:
+            ``ActorStarted``: Emitted once the actor has been started.
+
+        Args:
+            actor_spec: Spec for the actor to be started once resources are available.
+
+        Returns:
+            Actor info object that can be used to remove the actor or schedule futures
+                on it.
+
+        """
+        raise NotImplementedError
+
+    def remove_actor(
+        self,
+        actor_info: ActorInfo,
+        resolve_futures: bool = True,
+        kill: bool = False,
+    ) -> None:
+        """Remove an actor given its spec.
+
+        If the actor has already been started, this will remove the actor. If the
+        actor has only been requested, but not started, yet, this will cancel
+        the actor request.
+
+        If ``resolve_futures=True``, this will cache the actor removal and only
+        remove it once all its tracked futures are resolved.
+
+        If ``kill=True``, this will use ``ray.kill()`` to forcefully terminate the
+        actor. Otherwise, graceful actor deconstruction will be scheduled after
+        all currently tracked futures are resolved.
+
+        After stopping the actor, an ``ActorStopped`` event will be emitted.
+
+        Events:
+            ``ActorStopped``: Emitted once the actor has been started. If the actor
+                has only been requested (but has not been started, yet),
+                this event won't be emitted.
+
+        Args:
+            actor_info: Actor info object of the actor to be removed.
+            resolve_futures: If True, will resolve associated futures (and emit
+                events) first before stopping the actor.
+            kill: If set, will forcefully terminate the actor instead of gracefully
+                scheduling termination.
+        """
+        raise NotImplementedError
+
+    def is_actor_started(self, actor_info: ActorInfo) -> bool:
+        """Returns True if the actor has been started.
+
+        Args:
+            actor_info: Actor info object of the actor to inquire state about.
+        """
+        raise NotImplementedError
+
+    def get_actor_resources(self, actor_info: ActorInfo) -> Optional[AcquiredResources]:
+        """Returns the acquired resources of an actor that has been started.
+
+        This will return ``None`` if the actor has not been started, yet.
+
+        Args:
+            actor_info: Actor info object of the actor to get resources for.
+        """
+        raise NotImplementedError
+
+    def schedule_task(
+        self,
+        actor_info: ActorInfo,
+        task_spec: Tuple[str, Iterable[Any], Dict[str, Any]],
+        result_cls: Type[FutureResult],
+    ) -> None:
+        """Track (asynchronous) future associated to a tracked actor.
+
+        This method will schedule the remote task and track its execution. Once the
+        future is resolved, it will be emitted as an event. The event
+        class will be ``result_cls``, which should extend ``FutureResult``.
```

Comment:
do we have to batch future results ourselves?
it's usually a lot more efficient to handle multiple futures together.

Reply:
Is this only about the results (outputs) or also about scheduling (input)?

We could emit multiple results if the futures happen to resolve at the same time. For Tune that would actually complicate things a bit in processing as we do everything one-by-one and no batched processing.

For Train, where we actually do batched processing, we would do this with sync futures, i.e. futures that we require to resolve at the same time.

Maybe you can share a bit more how exactly rllib would like to interact with this. I.e. how are futures scheduled, how are they expected to resolve, and how are they processed.